code/datums/status_effects/debuffs/choke.dm

```diff
@@ -0,0 +1,300 @@
+	delta_x = 0
+	delta_y = 0
+
+/datum/status_effect/choke/proc/clear_flame()
+	flaming = FALSE
+	if(ash)
+		QDEL_NULL(ash)
+
+/datum/status_effect/choke/proc/vomit_up()
+	var/atom/movable/choking_on = choking_on_ref?.resolve()
+	if(choking_on && iscarbon(owner))
+		var/mob/living/carbon/carbon_owner = owner
+		// This will yeet the thing we're choking on out of us
+		carbon_owner.vomit(lost_nutrition = 20, force = TRUE, distance = 2)
+
+/datum/status_effect/choke/proc/on_vomit(mob/source, distance, force)
+	SIGNAL_HANDLER
+	var/atom/movable/choking_on = choking_on_ref?.resolve()
+	if(choking_on)
+		choking_on_ref = null
+		choking_on.forceMove(get_turf(source)) // Gotta be on a tile to throw yourself bestie
+		var/atom/target = get_edge_target_turf(source, source.dir)
+		choking_on.throw_at(target, distance, 1, source)
+
+/datum/status_effect/choke/get_examine_text()
+	return span_boldwarning("[owner.p_they(TRUE)] [owner.p_are()] choking!")
+
+/datum/status_effect/choke/proc/remove_choke(datum/source)
+	SIGNAL_HANDLER
+	if(!QDELETED(src))
+		qdel(src)
+
+/datum/status_effect/choke/proc/hazard_moved(atom/movable/source)
+	SIGNAL_HANDLER
+	if(QDELETED(src))
+		return
+	if(source.loc != owner)
+		qdel(src)
+
+/datum/status_effect/choke/proc/no_breathing(mob/living/source)
+	SIGNAL_HANDLER
+	RegisterSignal(source, SIGNAL_REMOVETRAIT(TRAIT_NOBREATH), .proc/on_breathable)
+	check_audio_state()
+
+/datum/status_effect/choke/proc/on_breathable(mob/living/source)
+	SIGNAL_HANDLER
+	UnregisterSignal(source, SIGNAL_REMOVETRAIT(TRAIT_NOBREATH))
+	check_audio_state()
+
+/datum/status_effect/choke/proc/on_logout(datum/source)
+	SIGNAL_HANDLER
+	clean_client()
+
+/datum/status_effect/choke/proc/on_death(mob/living/source)
+	SIGNAL_HANDLER
+	RegisterSignal(source, COMSIG_LIVING_REVIVE, .proc/on_revive)
+	check_audio_state()
+
+/datum/status_effect/choke/proc/on_revive(mob/living/source)
+	SIGNAL_HANDLER
+	UnregisterSignal(source, COMSIG_LIVING_REVIVE)
+	check_audio_state()
+
+/datum/status_effect/choke/proc/attempt_eat(mob/source, atom/eating)
+	SIGNAL_HANDLER
+	source.balloon_alert(source, "you can't get it down!")
+	return COMSIG_CARBON_BLOCK_EAT
+
+/datum/status_effect/choke/proc/helped(mob/source, mob/helping)
+	SIGNAL_HANDLER
+	INVOKE_ASYNC(src, .proc/heimlich, source, helping)
+	return COMPONENT_BLOCK_HELP_ACT
+
+/datum/status_effect/choke/proc/shook(mob/source, mob/helping)
+	SIGNAL_HANDLER
+	INVOKE_ASYNC(src, .proc/heimlich, source, helping)
+	return COMPONENT_BLOCK_MISC_HELP
+
+/datum/status_effect/choke/proc/heimlich(mob/victim, mob/aggressor)
+	if(victim == aggressor)
+		return
+	if(DOING_INTERACTION_WITH_TARGET(aggressor, victim))
+		victim.balloon_alert(aggressor, "You are already helping!")
+		return
+	if(DOING_INTERACTION(aggressor, "heimlich"))
+		victim.balloon_alert(aggressor, "You can't heimlich two people at once!") // Yes I know it's kinda runon, but this is a real rare condition
```

Comment:
```suggestion
		victim.balloon_alert(aggressor, "already helping someone!") // Yes I know it's kinda runon, but this is a real rare condition
```

Reply:
may as well remove the comment at that rate as well if "run-on" refers to the the text the balloon_alert shows